core/src/main/java/org/apache/calcite/sql/validate/implicit/TypeCoercionImpl.java

```diff
@@ -563,6 +564,44 @@ protected boolean booleanEquality(SqlCallBinding binding,
     return false;
   }
 
+  @Override public boolean quantifyOperationCoercion(SqlCallBinding binding) {
```

Comment:
is this only when the argument is an array?
If true, this should be documented.
(The function probably needs JavaDoc anyway)